backend/server/mailing.go

```diff
@@ -2,91 +2,239 @@ package main
+var mailjetClient *mailjet.Client
+
+// InitMailClient initialises a session with the Mailjet API and stores it in a global variable
+func InitMailClient() {
+	mailjetClient = mailjet.NewMailjetClient(publicKey, secretKey)
+
+	// Start mailing timers
+	go enquiryMailingTimer()
+	go feedbackMailingTimer()
 }
 
+///////////
+// HANDLERS
+///////////
+
 // HandleEnquiry by forwarding emails to relevant inboxes
 func HandleEnquiry(targetEmail string) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		message := Message{
+		// Extract fields from form
+		enquiry := Enquiry{
 			Name:  c.FormValue("name"),
 			Email: c.FormValue("email"),
 			Body:  c.FormValue("body"),
 		}
 
-		if err := c.Validate(message); err != nil {
+		// Validate struct
+		if err := c.Validate(enquiry); err != nil {
 			return c.JSON(http.StatusBadRequest, H{
 				"error": err,
 			})
 		}
 
-		// TODO: Compose and send email
-		return c.JSON(http.StatusOK, H{})
+		// Add to bundle
+		switch targetEmail {
+		case sponsorshipEmail:
+			sponsorshipBundle = append(sponsorshipBundle, enquiry)
+		case infoEmail:
+			infoBundle = append(infoBundle, enquiry)
+		}
+
+		return c.JSON(http.StatusAccepted, H{})
 	}
 }
 
-// FeedbackEnquiry - validates feedback input and forwards as an email to the revelant inbox
-func FeedbackEnquiry(targetEmail string) echo.HandlerFunc {
+// HandleFeedback - validates feedback input and forwards as an email to the revelant inbox
+func HandleFeedback() echo.HandlerFunc {
 	return func(c echo.Context) error {
-		name := c.FormValue("name")
-		email := c.FormValue("email")
-		body := c.FormValue("body")
-
-		// if there is no email, use the struct where emails are not required (FeedbackNoEmail)
-		if email == "" {
-			feedback := FeedbackNoEmail{
-				Name:  name,
-				Email: email,
-				Body:  body,
-			}
+		// Extract fields from form
+		feedback := Feedback{
+			Name:  c.FormValue("name"),
+			Email: c.FormValue("email"),
+			Body:  c.FormValue("body"),
+		}
 
-			if err := c.Validate(feedback); err != nil {
-				return c.JSON(http.StatusBadRequest, H{
-					"error": err,
-				})
-			}
+		// Validate struct
+		if err := c.Validate(feedback); err != nil {
+			return c.JSON(http.StatusBadRequest, H{
+				"error": err,
+			})
+		}
 
-			// if there is an email, check that the email is valid by using the Feedback Struct
-		} else {
-			feedback := Feedback{
-				Name:  c.FormValue("name"),
-				Email: c.FormValue("email"),
-				Body:  c.FormValue("body"),
-			}
+		// Add to bundle
+		feedbackBundle = append(feedbackBundle, feedback)
 
-			if err := c.Validate(feedback); err != nil {
-				return c.JSON(http.StatusBadRequest, H{
-					"error": err,
-				})
-			}
+		return c.JSON(http.StatusAccepted, H{})
+	}
+}
 
-		}
+/////////
+// TIMERS
+/////////
+
+// This function is executed once in a subroutine and triggers every 15 minutes
+func enquiryMailingTimer() {
+	for {
+		time.Sleep(15 * time.Minute)
+		DispatchEnquiryBundles()
```

Comment:
Lets check if the bundle is empty or not. I think its better to be defensive as possible

Reply:
I actually did the error checking in **sendEnquiryBundle** and **sendFeedbackBundle**. But I think it makes more sense to put it in **DispatchEnquiryBundles** and **DispatchFeedbackBundle** since those are the public wrapper functions.
Fixed in c5d61fc